aws/src/integration/java/org/apache/iceberg/aws/AwsIntegTestUtil.java

```diff
@@ -101,18 +103,20 @@ public static void cleanS3Bucket(S3Client s3, String bucketName, String prefix)
             ListObjectVersionsRequest.builder().bucket(bucketName).prefix(prefix).build());
     List<ObjectVersion> versionsToDelete = Lists.newArrayList();
     int batchDeletionSize = 1000;
-    response.versions().stream()
-        .forEach(
-            version -> {
-              versionsToDelete.add(version);
-              if (versionsToDelete.size() == batchDeletionSize) {
-                deleteObjectVersions(s3, bucketName, versionsToDelete);
-                versionsToDelete.clear();
-              }
-            });
-
-    if (!versionsToDelete.isEmpty()) {
-      deleteObjectVersions(s3, bucketName, versionsToDelete);
+    if (!checkIfS3Express()) {
```

Comment:
Why aren't we deleting the stuff here? At the very least we should be able to delete the objects that were created even without versioning